server/test/unit/server/controllers/test_repository.py

```diff
@@ -37,21 +37,21 @@ def setUp(self):
         ]
 
     def test_returns_ids(self, mock_objects):
-        mock_objects.return_value.only.return_value = self.associations
+        mock_objects.return_value.only.return_value.as_pymongo.return_value = self.associations
```

Comment:
I think you should change ``self.associations`` to have the correct new data type. Apparently model instances work correctly with dictionary-style attribute access, but that's not representative of the data that will actually flow through this function.